src/main/kotlin/org/rucca/cheese/notification/NotificationService.kt

```diff
@@ -1,112 +1,170 @@
-        notificationRepository.delete(notificationRepository.findById(notificationId).get())
+        val notification =
+            notificationRepository.findById(notificationId).orElseThrow {
+                NotFoundError("notification", notificationId)
+            }
+        notificationRepository.delete(notification)
     }
 
     fun markAsRead(notificationIds: List<Long>) {
         val notification = mutableListOf<Notification>()
-        for (id in notificationIds) {
-            val entity: Optional<Notification> = notificationRepository.findById(id)
-            if (entity.isPresent) {
-                notification.add(entity.get())
-            } else {
-                throw NotFoundError("notification", id)
-            }
+        for (notificationId in notificationIds) {
+            val entity =
+                notificationRepository.findByIdAndReceiver(
+                    notificationId,
+                    User().apply { id = authenticateService.getCurrentUserId().toInt() },
+                )
+            entity.ifPresentOrElse(
+                { notification.add(it) },
+                { throw NotFoundError("Notification not found", notificationId) },
+            )
         }
         notification.forEach { it.read = true }
         notificationRepository.saveAll(notification)
     }
 
     fun getUnreadCount(receiverId: Long): Int {
-        return notificationRepository.countByReceiverIdAndRead(receiverId, false).toInt()
+        return notificationRepository.countByReceiverAndRead(
+            User().apply { id = receiverId.toInt() },
+            false,
+        )
     }
 
     fun getNotificationOwner(notificationId: IdType): IdType {
         val notification = notificationRepository.findById(notificationId)
         if (!notification.isPresent) {
             throw NotFoundError("notification", notificationId)
         }
-        return notification.get().receiverId
+        return notification.get().receiver.id!!.toLong()
     }
 
-    fun isNotificationAdmin(notificationId: IdType, userId: IdType): Boolean {
-        val notification = notificationRepository.findByIdAndReceiverId(notificationId, userId)
-        if (!notification.isPresent) {
-            throw NotFoundError("notification", notificationId)
-        }
-        return notification.get().receiverId == authenticateService.getCurrentUserId()
+    fun isNotificationAdmin(notificationId: IdType?, userId: IdType): Boolean {
+        require(notificationId != null) { "notificationId cannot be null for this operation" }
+        val notification =
+            notificationRepository
+                .findByIdAndReceiver(notificationId, User().apply { id = userId.toInt() })
+                .orElseThrow { NotFoundError("notification", notificationId) }
+
+        return notification.receiver.id!!.toLong() == authenticateService.getCurrentUserId()
     }
 
     fun Notification.toNotificationDTO(notification: Notification): NotificationDTO {
```

Comment:
Well it's an extension method, which means you are "injecting" this function, into this the class, pretending that it's just like a normal method of that class. So Notification.toNotification is enough.
```suggestion
    fun Notification.toNotificationDTO(): NotificationDTO {
```

and don't forget to use `this` to refer to the caller in this method.